ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/domain.scala

```diff
@@ -235,7 +235,7 @@ object domain {
 
   sealed trait TransactionIdTag
 
-  type TransactionId = Ref.TransactionIdString @@ TransactionIdTag
+  type TransactionId = Ref.LedgerString @@ TransactionIdTag
   val TransactionId: Tag.TagOf[TransactionIdTag] = Tag.of[TransactionIdTag]
```

Comment:
Can't we keep this `TransactionId` type and get rid of the rest?

Reply:
It is not clear to me how to proceed, as I do not really know who (and how) is using the types defined in the following package:
- `com.digitalasset.ledger`
-`com.digitalasset.ledger.api.domain`
-`com.daml.ledger.participant.state.v1`


Furthermore to be consistent, we should also handle in similar way the types  `ParticipantId`,  `CommandId`, `WorkflowId`, `ApplicationId`. 

I think this exceed the scope of this PR.

Reply:
We use the types in `com.daml.ledger.participant.state.v1` for our `ReadService` and `WriteService` interfaces.